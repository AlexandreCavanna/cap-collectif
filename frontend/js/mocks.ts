/* eslint-env jest */
import type { IntlShape, MessageDescriptor } from 'react-intl'
import 'react-intl'
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay'

export const $refType: any = null
export const $fragmentRefs: any = null
export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: (message: MessageDescriptor) => String(message),
  formatPlural: (message: string) => String(message),
  formatNumber: (message: string) => String(message),
  formatRelative: (message: string) => String(message),
  formatTime: (message: string) => String(message),
  formatDate: (message: string) => String(message),
  formatMessage: (message: MessageDescriptor) => String(message.id),
}
export const formMock: ReduxFormFormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  // initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  updateSyncErrors: jest.fn(),
  valid: true,
  warning: null,
}
const environment = {
  UNSTABLE_getDefaultRenderPolicy: jest.fn(),
  __log: jest.fn(),
  getOperationTracker: jest.fn(),
  isRequestActive: jest.fn(),
  isServer: jest.fn(),
  options: jest.fn(),
  applyMutation: jest.fn(),
  sendMutation: jest.fn(),
  lookup: jest.fn(),
  sendQuery: jest.fn(),
  subscribe: jest.fn(),
  streamQuery: jest.fn(),
  retain: jest.fn(),
  unstable_internal: {
    areEqualSelectors: jest.fn(),
    createFragmentSpecResolver: jest.fn(),
    createOperationDescriptor: jest.fn(),
    getDataIDsFromFragment: jest.fn(),
    getDataIDsFromObject: jest.fn(),
    getFragment: jest.fn(),
    getPluralSelector: jest.fn(),
    getRequest: jest.fn(),
    getSelector: jest.fn(),
    getSelectorsFromObject: jest.fn(),
    getSingularSelector: jest.fn(),
    getVariablesFromFragment: jest.fn(),
    getVariablesFromObject: jest.fn(),
    getVariablesFromPluralFragment: jest.fn(),
    getVariablesFromSingularFragment: jest.fn(),
    isFragment: jest.fn(),
    isRequest: jest.fn(),
  },
  applyUpdate: jest.fn(),
  check: jest.fn(),
  commitPayload: jest.fn(),
  commitUpdate: jest.fn(),
  execute: jest.fn(),
  executeMutation: jest.fn(),
  executeWithSource: jest.fn(),
  getNetwork: jest.fn(),
  getStore: jest.fn(),
  areEqualSelectors: jest.fn(),
  createFragmentSpecResolver: jest.fn(),
  createOperationDescriptor: jest.fn(),
  getDataIDsFromFragment: jest.fn(),
  getDataIDsFromObject: jest.fn(),
  getFragment: jest.fn(),
  getPluralSelector: jest.fn(),
  getRequest: jest.fn(),
  getSelector: jest.fn(),
  getSelectorsFromObject: jest.fn(),
  getSingularSelector: jest.fn(),
  getVariablesFromFragment: jest.fn(),
  getVariablesFromObject: jest.fn(),
  getVariablesFromPluralFragment: jest.fn(),
  requiredFieldLogger: jest.fn(),
  revertUpdate: jest.fn(),
  replaceUpdate: jest.fn(),
}
export const relayPaginationMock: RelayPaginationProp = {
  environment,
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
}
export const relayRefetchMock: RelayRefetchProp = {
  environment,
  refetch: jest.fn(),
}
export const googleAddressMock = {
  lat: 43.6424564,
  lng: -79.3755156,
  json: '[{"formatted_address":"10 Yonge St, Toronto, ON M5E 1R4, Canada","geometry":{"location":{"lat":43.6424564,"lng":-79.3755156},"location_type":"ROOFTOP","viewport":{"south":43.6411074197085,"west":-79.37686458029151,"north":43.6438053802915,"east":-79.37416661970849}},"types":["street_address"],"address_components":[{"long_name":"10","short_name":"10","types":["street_number"]},{"long_name":"Yonge Street","short_name":"Yonge St","types":["route"]},{"long_name":"Old Toronto","short_name":"Old Toronto","types":["political","sublocality","sublocality_level_1"]},{"long_name":"Toronto","short_name":"Toronto","types":["locality","political"]},{"long_name":"Toronto Division","short_name":"Toronto Division","types":["administrative_area_level_2","political"]},{"long_name":"Ontario","short_name":"ON","types":["administrative_area_level_1","political"]},{"long_name":"Canada","short_name":"CA","types":["country","political"]},{"long_name":"M5E 1R4","short_name":"M5E 1R4","types":["postal_code"]}],"place_id":"ChIJrbgtxSvL1IkR2xGe7JZyuw0","plus_code":{"compound_code":"JJRF+XQ Toronto, ON, Canada","global_code":"87M2JJRF+XQ"}}]',
}
