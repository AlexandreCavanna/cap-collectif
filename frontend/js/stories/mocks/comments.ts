import { author } from './users'

export const comment = {
  id: 'comment1',
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2018',
  publishedAt: '1 mars 2018',
  published: true,
  pinned: true,
  body: 'Lorem ipsum dolor amet direct trade cornhole exercitation hashtag mixtape, master cleanse vexillologist. Irure kickstarter single-origin coffee fashion axe vexillologist viral. Selfies etsy cloud bread heirloom, sunt craft beer fixie palo santo asymmetrical.',
  votes: {
    totalCount: 50,
  },
  viewerHasVote: false,
  contribuable: true,
  reported: false,
}
export const comments = [
  { ...comment, id: 'comment1-2' },
  { ...comment, id: 'comment1-3', highlighted: true },
  { ...comment, id: 'comment1-7', trashedStatus: 'INVISIBLE', body: null },
  { ...comment, id: 'comment1-4', pinned: false },
  { ...comment, id: 'comment1-5', pinned: false, published: false },
  { ...comment, id: 'comment1-6', user: { ...comment.user, isViewer: true } },
]
