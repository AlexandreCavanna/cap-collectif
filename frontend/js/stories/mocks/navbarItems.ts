export const items = [
  {
    id: 12,
    title: 'Accueil',
    link: '/',
    hasEnabledFeature: true,
    children: [],
    active: true,
  },
  {
    id: 5,
    title: 'Actualités',
    link: '/blog',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 11,
    title: 'Thèmes',
    link: '/themes',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 4,
    title: 'Évènements',
    link: '/events',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 10,
    title: 'Projets participatifs',
    link: '/projects',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 23,
    title: 'Custom page',
    link: '/custom-page',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
  {
    id: 6,
    title: 'Comment ça marche',
    link: '/pages/comment-%C3%A7a-marche',
    hasEnabledFeature: true,
    children: [],
    active: false,
  },
]
export const itemWithChildren = {
  id: 9,
  title: 'À propos',
  link: '',
  hasEnabledFeature: true,
  children: [
    {
      id: 8,
      title: 'Contact',
      link: '/contact',
      hasEnabledFeature: true,
      active: false,
    },
    {
      id: 3,
      title: 'Liste des inscrits',
      link: '/members',
      hasEnabledFeature: true,
      active: false,
    },
    {
      id: 2,
      title: 'Lien externe',
      link: '/www.google.fr',
      hasEnabledFeature: true,
      active: false,
    },
  ],
  active: false,
}
