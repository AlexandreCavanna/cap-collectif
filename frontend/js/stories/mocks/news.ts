import { author } from './users'

export const post = {
  title: 'Je suis le titre',
  url: '#',
  image: {
    src: 'https://source.unsplash.com/random/150x100',
    alt: 'news picture',
  },
  body: ' Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  date: {
    released: '2019-11-09',
  },
  theme: 'Test & Democratie',
  user: author,
}
export const withoutParaph = {
  title: 'un titre pour le deuxi√®me cas',
  url: '#',
  image: {
    src: 'https://source.unsplash.com/random/150x100',
    alt: 'visual for other case',
  },
  date: {
    released: '19/12/2019',
  },
  theme: 'Test et Jest',
  user: author,
  body: '',
}
export default post
