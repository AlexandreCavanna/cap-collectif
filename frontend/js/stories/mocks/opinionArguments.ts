import { author } from './users'

export const opinionArgument = {
  id: 1,
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  published: true,
  related: {
    url: 'https://ui.cap-collectif.com/',
    title: 'Opinion XXX',
  },
  body: 'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: {
    totalCount: 50,
  },
  viewerHasVote: false,
  contribuable: true,
  reported: false,
}
export const opinionArguments = [
  { ...opinionArgument, id: 1 },
  { ...opinionArgument, id: 2, user: { ...opinionArgument.user, vip: true } },
  { ...opinionArgument, id: 3, user: null },
  { ...opinionArgument, id: 4, trashedStatus: 'INVISIBLE', body: null },
  { ...opinionArgument, id: 5, published: false },
  { ...opinionArgument, id: 6, viewerHasVote: true },
  { ...opinionArgument, id: 7, contribuable: false },
  { ...opinionArgument, id: 8, viewerHasVote: true, contribuable: false },
  { ...opinionArgument, id: 9, user: { ...opinionArgument.user, isViewer: true } },
  { ...opinionArgument, id: 10, reported: true },
]
