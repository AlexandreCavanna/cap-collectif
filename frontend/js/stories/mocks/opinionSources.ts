import { author } from './users'

export const opinionSource = {
  url: 'https://ui.cap-collectif.com/',
  trashedStatus: '',
  user: author,
  createdAt: ' • 1 mars 2016',
  publishedAt: ' • 1 mars 2016',
  title: 'Source title',
  category: 'Politique',
  related: {
    url: 'https://ui.cap-collectif.com/',
    title: 'Opinion XXX',
  },
  body: 'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: {
    totalCount: 2,
  },
  viewerHasVote: false,
  contribuable: true,
  reported: false,
}
export const opinionSources = [
  { ...opinionSource },
  { ...opinionSource, user: { ...opinionSource.user, vip: true } },
  { ...opinionSource, category: null },
  { ...opinionSource, trashedStatus: 'INVISIBLE' },
  { ...opinionSource, viewerHasVote: true },
  { ...opinionSource, contribuable: false },
  { ...opinionSource, viewerHasVote: true, contribuable: false },
  { ...opinionSource, user: { ...opinionSource.user, isViewer: true } },
  { ...opinionSource, reported: true },
]
