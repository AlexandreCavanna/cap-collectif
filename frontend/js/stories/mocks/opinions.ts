import { author } from './users'

export const opinion = {
  id: 'opinion1',
  url: 'https://ui.cap-collectif.com/',
  title: 'Opinion',
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2018',
  publishedAt: '1 mars 2018',
  published: true,
  pinned: false,
  votes: {
    totalCount: 8,
  },
  versions: {
    totalCount: 3,
  },
  arguments: {
    totalCount: 4,
  },
  sources: {
    totalCount: 0,
  },
  votesMitige: {
    totalCount: 1,
  },
  votesNok: {
    totalCount: 3,
  },
  votesOk: {
    totalCount: 4,
  },
  previewVotes: new Array(8).fill(author),
  body: 'Lorem ipsum dolor amet direct trade cornhole exercitation hashtag mixtape, master cleanse vexillologist. Irure kickstarter single-origin coffee fashion axe vexillologist viral. Selfies etsy cloud bread heirloom, sunt craft beer fixie palo santo asymmetrical.',
  project: {
    opinionCanBeFollowed: true,
  },
  section: {
    voteWidgetType: 2,
  },
  appendices: [],
  answer: null,
}
export const opinions = [
  { ...opinion },
  { ...opinion, user: { ...opinion.user, vip: true } },
  { ...opinion, user: null },
  {
    ...opinion,
    votes: {
      totalCount: 0,
    },
  },
  { ...opinion, updatedAt: '15 mars 2018' },
  { ...opinion, updatedAt: '15 mars 2018', pinned: true },
  { ...opinion, updatedAt: '15 mars 2018', ranking: 'Label' },
  { ...opinion, updatedAt: '15 mars 2018', pinned: true, ranking: 'Label' },
  { ...opinion, ranking: 'Label' },
  { ...opinion, trashedStatus: 'INVISIBLE' },
  { ...opinion, pinned: true, ranking: 'Label' },
  { ...opinion, pinned: true },
  { ...opinion, published: false },
]
