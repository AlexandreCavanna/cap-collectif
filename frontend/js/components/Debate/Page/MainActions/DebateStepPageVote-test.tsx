/* eslint-env jest */
import * as React from 'react'
import { render } from 'enzyme'
import { DebateStepPageVote } from './DebateStepPageVote'
import { $refType } from '~/mocks'
import { MockProviders } from '~/testUtils'

const baseProps = {
  step: {
    ' $refType': $refType,
    isAnonymousParticipationAllowed: false,
    debate: {
      url: '/debate123',
      id: 'debate-123',
      yesVotes: {
        totalCount: 12,
      },
      votes: {
        totalCount: 20,
      },
    },
  },
}
const props = {
  basic: baseProps,
  anonymousParticipationAllowed: { ...baseProps, step: { ...baseProps.step, isAnonymousParticipationAllowed: true } },
}
const normalUser = {
  user: {
    isEmailConfirmed: true,
  },
}
const nonConfirmedUser = {
  user: {
    isEmailConfirmed: false,
  },
}
const nonAuthenticated = {
  user: null,
}
describe('<DebateStepPageVote/>', () => {
  it('should renders correctly', () => {
    const wrapper = render(
      <MockProviders
        store={{
          user: normalUser,
        }}
        useCapUIProvider
      >
        <DebateStepPageVote {...props.basic} />
      </MockProviders>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should renders correctly when not authenticated', () => {
    const wrapper = render(
      <MockProviders
        store={{
          user: nonAuthenticated,
        }}
        useCapUIProvider
      >
        <DebateStepPageVote {...props.basic} />
      </MockProviders>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should renders correctly with email not confirmed argument', () => {
    const wrapper = render(
      <MockProviders
        store={{
          user: nonConfirmedUser,
        }}
        useCapUIProvider
      >
        <DebateStepPageVote {...props.basic} />
      </MockProviders>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should renders correctly with anonymous participation', () => {
    const wrapper = render(
      <MockProviders
        store={{
          user: normalUser,
        }}
        useCapUIProvider
      >
        <DebateStepPageVote {...props.anonymousParticipationAllowed} />
      </MockProviders>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
