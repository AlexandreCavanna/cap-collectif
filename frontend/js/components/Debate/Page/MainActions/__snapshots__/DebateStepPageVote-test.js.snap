// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebateStepPageVote/> should renders correctly 1`] = `
<div
  class="sc-bdnxRM gmUUoo"
  style="opacity:0"
>
  <button
    class="sc-bdnxRM omwMO"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        fill="none"
        viewBox="0 0 32 29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 13.09a3.334 3.334 0 00-3.333-3.333h-8.415a.333.333 0 01-.316-.437 11.58 11.58 0 00.693-5.876c-.784-2.8-2.445-3.6-3.744-3.42a2.667 2.667 0 00-2.218 2.733c0 3.662-2.839 7.662-5.379 9.415a.666.666 0 00-.288.55v13.487a.667.667 0 00.54.655c.476.09.908.18 1.316.267 1.794.427 3.633.637 5.477.628H25c2.61 0 3-1.467 3-2.334 0-.46-.105-.914-.308-1.326a3.001 3.001 0 001.333-4 3 3 0 001.279-4.104A3.334 3.334 0 0032 13.09zM7 12.757a1.333 1.333 0 00-1.333-1.333H1.333A1.333 1.333 0 000 12.757v14.667a1.333 1.333 0 001.333 1.333h4.334A1.334 1.334 0 007 27.424V12.757zM5 25.091a1 1 0 11-2 0 1 1 0 012 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.for
      </p>
    </div>
  </button>
  <button
    class="sc-bdnxRM fpzkOi"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        viewBox="0 0 12 11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 5.875a1.25 1.25 0 001.25 1.25h3.155a.125.125 0 01.119.164 4.342 4.342 0 00-.26 2.204c.294 1.05.917 1.35 1.404 1.282A1 1 0 006.5 9.75c0-1.373 1.064-2.873 2.017-3.53a.25.25 0 00.108-.206V.956A.25.25 0 008.422.71c-.178-.034-.34-.067-.493-.1A8.68 8.68 0 005.875.375h-3.25C1.646.375 1.5.923 1.5 1.25c0 .173.039.344.115.5a1.125 1.125 0 00-.5 1.5 1.125 1.125 0 00-.479 1.539A1.25 1.25 0 000 5.875zM9.375 6a.5.5 0 00.5.5H11.5A.5.5 0 0012 6V.5a.5.5 0 00-.5-.5H9.875a.5.5 0 00-.5.5V6zm.75-4.625a.375.375 0 11.75 0 .375.375 0 01-.75 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.against
      </p>
    </div>
  </button>
</div>
`;

exports[`<DebateStepPageVote/> should renders correctly when not authenticated 1`] = `
<div
  class="sc-bdnxRM gmUUoo"
  style="opacity:0"
>
  <button
    aria-controls="popover-overlay"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="sc-bdnxRM omwMO"
    type="button"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        fill="none"
        viewBox="0 0 32 29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 13.09a3.334 3.334 0 00-3.333-3.333h-8.415a.333.333 0 01-.316-.437 11.58 11.58 0 00.693-5.876c-.784-2.8-2.445-3.6-3.744-3.42a2.667 2.667 0 00-2.218 2.733c0 3.662-2.839 7.662-5.379 9.415a.666.666 0 00-.288.55v13.487a.667.667 0 00.54.655c.476.09.908.18 1.316.267 1.794.427 3.633.637 5.477.628H25c2.61 0 3-1.467 3-2.334 0-.46-.105-.914-.308-1.326a3.001 3.001 0 001.333-4 3 3 0 001.279-4.104A3.334 3.334 0 0032 13.09zM7 12.757a1.333 1.333 0 00-1.333-1.333H1.333A1.333 1.333 0 000 12.757v14.667a1.333 1.333 0 001.333 1.333h4.334A1.334 1.334 0 007 27.424V12.757zM5 25.091a1 1 0 11-2 0 1 1 0 012 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.for
      </p>
    </div>
  </button>
  <div
    aria-label="vote.popover.title"
    class="sc-bdnxRM NewLoginOverlay__PopoverContainer-hazOKr gA-dolT ehiDfV"
    data-dialog="true"
    hidden=""
    id="login-popover"
    role="dialog"
    style="display:none;position:fixed;left:100%;top:100%"
    tabindex="0"
  >
    <div
      class="sc-bdnxRM NewLoginOverlay__Arrow-csxmwg GwkjI ipxCrt"
      style="font-size:30px;width:1em;height:1em;pointer-events:none;bottom:100%"
    >
      <svg
        style="transform:rotateZ(360deg)"
        viewBox="0 0 30 30"
      >
        <path
          class="stroke"
          d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
        />
        <path
          class="fill"
          d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
        />
      </svg>
    </div>
    <div
      class="sc-bdnxRM jgfvhW"
    >
      <p
        class="sc-bdnxRM hdVRdr cap-text"
      >
        vote.popover.title
      </p>
      <div
        class="sc-bdnxRM jRHUGW"
      >
        <p
          class="sc-bdnxRM bhbXgN cap-text"
          style="margin-left:0;margin-top:0"
        >
          vote.popover.body
        </p>
        <span
          class="sc-bdnxRM hptjof"
          clip="rect(0px, 0px, 0px, 0px)"
        >
          <button
            class="sc-bdnxRM sc-lbVvki juQjuI fKaghU cap-button"
            type="button"
          >
            global.close
          </button>
        </span>
        <span>
          <button
            aria-label="open.connection_modal"
            class="center-block btn-block login-button btn btn-success"
            destination="http://localhost"
            type="button"
          >
            global.login
          </button>
        </span>
      </div>
    </div>
  </div>
  <button
    aria-controls="popover-overlay"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="sc-bdnxRM fpzkOi"
    type="button"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        viewBox="0 0 12 11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 5.875a1.25 1.25 0 001.25 1.25h3.155a.125.125 0 01.119.164 4.342 4.342 0 00-.26 2.204c.294 1.05.917 1.35 1.404 1.282A1 1 0 006.5 9.75c0-1.373 1.064-2.873 2.017-3.53a.25.25 0 00.108-.206V.956A.25.25 0 008.422.71c-.178-.034-.34-.067-.493-.1A8.68 8.68 0 005.875.375h-3.25C1.646.375 1.5.923 1.5 1.25c0 .173.039.344.115.5a1.125 1.125 0 00-.5 1.5 1.125 1.125 0 00-.479 1.539A1.25 1.25 0 000 5.875zM9.375 6a.5.5 0 00.5.5H11.5A.5.5 0 0012 6V.5a.5.5 0 00-.5-.5H9.875a.5.5 0 00-.5.5V6zm.75-4.625a.375.375 0 11.75 0 .375.375 0 01-.75 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.against
      </p>
    </div>
  </button>
  <div
    aria-label="vote.popover.title"
    class="sc-bdnxRM NewLoginOverlay__PopoverContainer-hazOKr gA-dolT ehiDfV"
    data-dialog="true"
    hidden=""
    id="login-popover"
    role="dialog"
    style="display:none;position:fixed;left:100%;top:100%"
    tabindex="0"
  >
    <div
      class="sc-bdnxRM NewLoginOverlay__Arrow-csxmwg GwkjI ipxCrt"
      style="font-size:30px;width:1em;height:1em;pointer-events:none;bottom:100%"
    >
      <svg
        style="transform:rotateZ(360deg)"
        viewBox="0 0 30 30"
      >
        <path
          class="stroke"
          d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
        />
        <path
          class="fill"
          d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
        />
      </svg>
    </div>
    <div
      class="sc-bdnxRM jgfvhW"
    >
      <p
        class="sc-bdnxRM hdVRdr cap-text"
      >
        vote.popover.title
      </p>
      <div
        class="sc-bdnxRM jRHUGW"
      >
        <p
          class="sc-bdnxRM bhbXgN cap-text"
          style="margin-left:0;margin-top:0"
        >
          vote.popover.body
        </p>
        <span
          class="sc-bdnxRM hptjof"
          clip="rect(0px, 0px, 0px, 0px)"
        >
          <button
            class="sc-bdnxRM sc-lbVvki juQjuI fKaghU cap-button"
            type="button"
          >
            global.close
          </button>
        </span>
        <span>
          <button
            aria-label="open.connection_modal"
            class="center-block btn-block login-button btn btn-success"
            destination="http://localhost"
            type="button"
          >
            global.login
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DebateStepPageVote/> should renders correctly with anonymous participation 1`] = `
<div
  class="sc-bdnxRM gmUUoo"
  style="opacity:0"
>
  <button
    class="sc-bdnxRM omwMO"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        fill="none"
        viewBox="0 0 32 29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 13.09a3.334 3.334 0 00-3.333-3.333h-8.415a.333.333 0 01-.316-.437 11.58 11.58 0 00.693-5.876c-.784-2.8-2.445-3.6-3.744-3.42a2.667 2.667 0 00-2.218 2.733c0 3.662-2.839 7.662-5.379 9.415a.666.666 0 00-.288.55v13.487a.667.667 0 00.54.655c.476.09.908.18 1.316.267 1.794.427 3.633.637 5.477.628H25c2.61 0 3-1.467 3-2.334 0-.46-.105-.914-.308-1.326a3.001 3.001 0 001.333-4 3 3 0 001.279-4.104A3.334 3.334 0 0032 13.09zM7 12.757a1.333 1.333 0 00-1.333-1.333H1.333A1.333 1.333 0 000 12.757v14.667a1.333 1.333 0 001.333 1.333h4.334A1.334 1.334 0 007 27.424V12.757zM5 25.091a1 1 0 11-2 0 1 1 0 012 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.for
      </p>
    </div>
  </button>
  <button
    class="sc-bdnxRM fpzkOi"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        viewBox="0 0 12 11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 5.875a1.25 1.25 0 001.25 1.25h3.155a.125.125 0 01.119.164 4.342 4.342 0 00-.26 2.204c.294 1.05.917 1.35 1.404 1.282A1 1 0 006.5 9.75c0-1.373 1.064-2.873 2.017-3.53a.25.25 0 00.108-.206V.956A.25.25 0 008.422.71c-.178-.034-.34-.067-.493-.1A8.68 8.68 0 005.875.375h-3.25C1.646.375 1.5.923 1.5 1.25c0 .173.039.344.115.5a1.125 1.125 0 00-.5 1.5 1.125 1.125 0 00-.479 1.539A1.25 1.25 0 000 5.875zM9.375 6a.5.5 0 00.5.5H11.5A.5.5 0 0012 6V.5a.5.5 0 00-.5-.5H9.875a.5.5 0 00-.5.5V6zm.75-4.625a.375.375 0 11.75 0 .375.375 0 01-.75 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.against
      </p>
    </div>
  </button>
</div>
`;

exports[`<DebateStepPageVote/> should renders correctly with email not confirmed argument 1`] = `
<div
  class="sc-bdnxRM gmUUoo"
  style="opacity:0"
>
  <button
    class="sc-bdnxRM omwMO"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        fill="none"
        viewBox="0 0 32 29"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 13.09a3.334 3.334 0 00-3.333-3.333h-8.415a.333.333 0 01-.316-.437 11.58 11.58 0 00.693-5.876c-.784-2.8-2.445-3.6-3.744-3.42a2.667 2.667 0 00-2.218 2.733c0 3.662-2.839 7.662-5.379 9.415a.666.666 0 00-.288.55v13.487a.667.667 0 00.54.655c.476.09.908.18 1.316.267 1.794.427 3.633.637 5.477.628H25c2.61 0 3-1.467 3-2.334 0-.46-.105-.914-.308-1.326a3.001 3.001 0 001.333-4 3 3 0 001.279-4.104A3.334 3.334 0 0032 13.09zM7 12.757a1.333 1.333 0 00-1.333-1.333H1.333A1.333 1.333 0 000 12.757v14.667a1.333 1.333 0 001.333 1.333h4.334A1.334 1.334 0 007 27.424V12.757zM5 25.091a1 1 0 11-2 0 1 1 0 012 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.for
      </p>
    </div>
  </button>
  <button
    class="sc-bdnxRM fpzkOi"
  >
    <div
      class="sc-bdnxRM IoRUe"
    >
      <svg
        class="sc-bdnxRM sc-eJocfa AbwPc kuPXuU cap-icon"
        viewBox="0 0 12 11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 5.875a1.25 1.25 0 001.25 1.25h3.155a.125.125 0 01.119.164 4.342 4.342 0 00-.26 2.204c.294 1.05.917 1.35 1.404 1.282A1 1 0 006.5 9.75c0-1.373 1.064-2.873 2.017-3.53a.25.25 0 00.108-.206V.956A.25.25 0 008.422.71c-.178-.034-.34-.067-.493-.1A8.68 8.68 0 005.875.375h-3.25C1.646.375 1.5.923 1.5 1.25c0 .173.039.344.115.5a1.125 1.125 0 00-.5 1.5 1.125 1.125 0 00-.479 1.539A1.25 1.25 0 000 5.875zM9.375 6a.5.5 0 00.5.5H11.5A.5.5 0 0012 6V.5a.5.5 0 00-.5-.5H9.875a.5.5 0 00-.5.5V6zm.75-4.625a.375.375 0 11.75 0 .375.375 0 01-.75 0z"
          fill="currentColor"
        />
      </svg>
      <p
        class="sc-bdnxRM ipdWIi cap-text"
      >
        global.against
      </p>
    </div>
  </button>
</div>
`;
